apps/embed-iframe/src/EmbeddedComponent.tsx

```diff
@@ -34,8 +33,22 @@ export function EmbeddedComponent() {
       (window as any).attachEvent("onmessage", handleRequest);
     }
 
-    sendResponse({ type: "init_complete" });
-  });
+    if (getUserData() !== null) {
+      sendResponse({ type: "init_complete", userData: getUserData()! });
+    } else {
+      sendResponse({ type: "init_complete" });
+    }
+
+    return () => {
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      if (window.removeEventListener) {
+        window.removeEventListener("message", handleRequest, false);
+      } else {
+        // fallback for older browsers (IE8 and below)
+        (window as any).detachEvent("onmessage", handleRequest);
```

Comment:
maybe it's worth checking that this function is present?

Reply:
Fixed in previous PR: https://github.com/trilitech/umami-v2/pull/1500/commits/ca778852ca9b7519c4cf98582402b2112824464f